src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -465,7 +465,25 @@ private void AddDirectives(SyntaxNode node, TextSpan span)
 
                         if (_directivesToKeep.Contains(directive))
                         {
-                            AddResidualTrivia(SyntaxFactory.TriviaList(directive.ParentTrivia), requiresNewLine: true);
+                            var parentTrivia = directive.ParentTrivia;
+                            var parentToken = parentTrivia.Token;
+
+                            var triviaList = parentToken.LeadingTrivia.Contains(parentTrivia)
```

Comment:
It feels like just an `IndexOf` check would be sufficient here. `-1` is returned for not found, and as far as I know it does similar amount of work as a `Contains` check. So in the case the directive is in a leading trivia we avoid an extra search of the trivia list.

Basically just check `LeadingTrivia.IndexOf` and if you get a `-1` out then do the same check for `TrailingTrivia`.

`var triviaList = parentToken.LeadingTrivia.IndexOf(parentTrivia) is not -1 and var index1 ? index1 : parentToken.TrailingTrivia.IndexOf(parentTrivia);`

Reply:
done.